File: 2주차/송성욱/조이스틱.cpp
#include <string>
#include <vector>
#include <algorithm>
 
using namespace std;
 
int solution(string name) {
    int answer = 0; 
    
    //A변경 순방향 이동 역방향 이동 
    for(auto ch: name) answer += min(ch - 'A', 91 - ch);
 
    int len = name.length();
    int move = len - 1;
    int next;
    
    for(int i = 0; i < len; i++) {
        next = i + 1;
        //a연속시 처리
        while(next < len && name[next] == 'A') next++;
        
        move = min(move, i + (len - next) + min(i, len - next));
    }
    answer += move;
 
    return answer;
}

File: 2주차/김태경/조이스틱.java
class Solution {
    int answer = 0;
    
    public int solution(String name) {
        run(name.toCharArray()); 
        return answer;
    }
    
    void run(char[] name) {
        int length = name.length;
        char[] now = new char[length];
        
        for (int i = 0; i < length; i++) {
            now[i] = 'A';
        }
        
        int move = length - 1;  
        
        for(int idx=0;idx<length;idx++){
            if (now[idx] != name[idx]) {
                int up = name[idx] - 'A';  
                int down = 'Z' - name[idx] + 1;  
                answer += Math.min(up, down); 
                now[idx] = name[idx];  
            }
            

            
            int index = idx + 1;  
            while (index < length && name[index] == 'A') {
                index++;
            }
            move = Math.min(move, idx * 2 + length - index);
            move = Math.min(move, (length - index) * 2 + idx);
            
        }
        
        answer += move;
    }
}
